### PR TITLE
nes.xml: Two redumps, two new dumps, one dump removed.

### PR DESCRIPTION
**Pull request by 0kmg**
- Replaced smb10 and smario14 with cart redumps. Previous dumps were rom hacks to allow the games to work on ancient emulators. [NewRisingSun]
- Likewise removed rom hack smb11a. smb11 is the proper pirate MMC3-variant dump, so it is retained.
- Hooked up heromali to correct slot device. Works now.
- Hooked smb4 up to correct slot. Works now as well.
- Bonus: hooked magicblks up to a proper slot device. Fixes graphics.

New working software list additions
-----------------------------------
The Ancient Modern Heros [NewRisingSun]
Super Mari - Mali Lugi - Mari no Daisakusen 2 [NewRisingSun]

Software list items promoted to working
---------------------------------------
Super Mario Bros. IV (Armadillo pirate)

### CODE REVIEW

File: hash/nes.xml [hunk 1]
```diff
@@ -73901,13 +73918,13 @@ Other
...
-			<feature name="slot" value="txrom" />
-			<feature name="pcb" value="NES-TLROM" />
+			<feature name="slot" value="sbros11" />
+			<feature name="pcb" value="BTL-SUPERBROS11" />
 			<dataarea name="chr" size="131072">
 				<rom name="super bros 10 kung fu mari (jackie chan hack).chr" size="131072" crc="b0382d58" sha1="d754d208564573fe5ea66111595f3b7e76cd2f33" offset="00000" status="baddump" />
 			</dataarea>
 			<dataarea name="prg" size="131072">
-				<rom name="super bros 10 kung fu mari (jackie chan hack).prg" size="131072" crc="43e1ec79" sha1="cabc927a5693523ada990adcf4f82227c43c8c64" offset="00000" status="baddump" />
+				<rom name="super bros 10 kung fu mari (jackie chan hack).prg" size="131072" crc="dc37d3db" sha1="8fcff188ca9425859bc7463689604a515139b6b1" offset="00000" status="baddump" />
```
Comment (cuavas) [inline hash/nes.xml:73927]: Is this still a bad dump?

File: hash/nes.xml [hunk 2]
```diff
@@ -74053,13 +74070,13 @@ Other
...
-			<feature name="slot" value="txrom" />
-			<feature name="pcb" value="NES-TLROM" />
+			<feature name="slot" value="sbros11" />
+			<feature name="pcb" value="BTL-SUPERBROS11" />
 			<dataarea name="chr" size="131072">
 				<rom name="super mario 14 (unl).chr" size="131072" crc="c54e1415" sha1="59e4262c2a67655d11c0b839e29cd46a2f41655a" offset="00000" status="baddump" />
 			</dataarea>
 			<dataarea name="prg" size="131072">
-				<rom name="super mario 14 (unl).prg" size="131072" crc="d838a08d" sha1="f55cd39674eacc72ffdcb5fe4ec4d1a4fac023ab" offset="00000" status="baddump" />
+				<rom name="super mario 14 (unl).prg" size="131072" crc="f17d41f2" sha1="528891aaada0ebd4e57257bd928e89d105760ea5" offset="00000" status="baddump" />
```
Comment (cuavas) [inline hash/nes.xml:74079]: Is this still a bad dump?

File: hash/nes.xml (hunk 1 above)
Comment (0kmg) [reply hash/nes.xml:73927]: Yeah, I suppose I need direction on this. It seems about half of nes.xml is marked as a bad dump. There's a comment about half way down the file after the initial confirmed dumps section that says:

> <!-- The following dumps are marked as bad dumps because they have been obtained by splitting PRG/CHR from NoIntro images and by using as PCB the ones corresponding to the mapper present in the iNES header. They are probably good dumps in most of the case, but for the documentation to be 100% correct we would like some dumper to verify both the correctness of the dumps and which PCB is present inside the cart.

Pretty much everything after this point (and most things I've included lately) is following this guideline and marking things as "baddump". The dumps here are typical instances—the dumpers release things in .nes format, I take it, strip off the .nes header, and split the file based on PRG/CHR size indicted in the header.

TL;DR: I've been extremely shy about either removing "baddump" flags or adding anything without "baddump" flags.

File: hash/nes.xml (hunk 2 above)
Comment (0kmg) [reply hash/nes.xml:74079]: See previous comment.

File: hash/nes.xml (hunk 1 above)
Comment (DavidHaywood) [reply hash/nes.xml:73927]: this does however send out a wrong message.

outside of the MAME community the MAME list is seen as inferior, as it's just "full of bad dumps" (in part because we're saying they're bad)  

I've even seen one place *ban* the MAME NES sets, because they're 'just bad dumps' (while continuing to host the old 'goodtools' junk)

Comment (0kmg) [reply hash/nes.xml:73927]: Yes, that's kind of why I said I need direction. The NES software list is just about the only one to liberally apply "baddump" to half its entries. A "provisional" or whatnot flag could help—though it would be mostly cosmetic as we'd more or less search and replace baddump flags.

The other option is restructuring the list. The SNES list for instance just keeps all the unconfirmed/unsorted stuff at the end. The NES list in some sense also already does this with its second half where all the baddump flags are. A big difference however is that NES's second half is more-or-less Asian/Russian markets and/or bootlegs/pirates. There's no "confirmed" section for things to be promoted to because the first half of nes.xml is only games, licensed or not (Tengen etc), from official regions. I don't particularly like this option either as the second half of the list is further subdivided into subsections for various companies, multicarts, etc (I certainly don't want to maintain pairs of confirmed and unconfirmed subsections). Of course in practice since virtually nothing is confirmed for Asian market bootlegs etc this second option would more or less do away with baddump flags and everything would be in unconfirmed sections.
